### ISSUE TITLE
$request->user() is always null on canSee when using Metric with refreshWhenFiltersChange

### ISSUE THREAD

**Issue by tonyputi** [laravel/nova-issues#5181]
- Laravel Version: 9.45.1
- Nova Version: 4.19.5
- PHP Version: 8.1.13
- Database Driver & Version: MySQL 8.0 / MariaDB 10.0
- Operating System and Version: Ventura 13.1
- Browser type and version: Chome 108.0.5359.124
- Reproduction Repository: https://github.com/###/###

### Description:
When using Metric with option `refreshWhenFiltersChange()` and a field of the same resource is make use of `canSee(fn ($request) => $request->user()->is_admin)` an error `Attempt to read property "is_admin" on null` is thrown.

### Detailed steps to reproduce the issue on a fresh Nova installation:

- Create a vanilla Laravel with Nova installation;
- Create whatever resource in my case I used the default Nova User Resource;
- Make use of `canSee` on one of the fields trying to access whatever attribute of the currently logged-in user. For instance: 
```
Text::make('Name')
    ->sortable()
    ->canSee(fn ($request) => $request->user()->is_admin)
    ->rules('required', 'max:255'),
```
- Create a new metric ie `php artisan nova:partition UsersPerPlan`
- Define the newly created Metric into the User resource cards method and make use of `refreshWhenFiltersChange` 
```
/**
 * Get the cards available for the request.
 *
 * @param  \Laravel\Nova\Http\Requests\NovaRequest  $request
 * @return array
 */
public function cards(NovaRequest $request)
{
    return [
        Metrics\UsersPerPlan::make()
            ->refreshWhenFiltersChange()
    ];
}
```
- Reload the page and the error `Attempt to read property "is_admin" on null` is thrown.

Basically, it is impossible to access the `$request->user()` that results always null when a `MetricRequest` is processed.

The problem looks like related to the file: `vendor/laravel/nova/src/Http/Requests/MetricRequest.php` more specifically on methods `availableMetrics` and  `availableMetricsForDetail` on the calls:
`ResourceDetailRequest::createFromBase($this)`, `ResourceIndexRequest::createFromBase($this)`

I hope to help you guys easily find the issue and feel free to contact me for further informations.

Filippo Sallemi
